### PR TITLE
FreeIPA: new module ipa_config

### PR DESCRIPTION
**Pull request by fxfitz**
##### SUMMARY

I recently came across the need to modify some global settings in FreeIPA, so this module allows us to do that.

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

ipa_config

##### ANSIBLE VERSION

```
ansible 2.7.0.dev0 (ipa_config_module df4dbc9ace) last updated 2018/07/03 16:30:19 (GMT -500)
  config file = None
  configured module search path = [u'/Users/fxfitz/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/fxfitz/dev/ansible/lib/ansible
  executable location = /Users/fxfitz/.pyenv/versions/ansible/bin/ansible
  python version = 2.7.15 (default, May 29 2018, 20:16:38) [GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.1)]
```

##### ADDITIONAL INFORMATION

Right now I have it set up only to modify the `ipadefaultloginshell` and `ipadefaultloginshell` keys of the API, but I think I want to add all of the possible options, I just haven't done that yet.

Additionally, I want to make sure the module arguments match up with the other FreeIPA modules as best as possible.  For example, in `ipa_user` the login shell argument is just `loginshell`, so I think we should probably set aliases so the modules have the same "feel".

```

```

### CODE REVIEW

File: test/sanity/validate-modules/ignore.txt [hunk 1]
```diff
@@ -617,6 +617,7 @@ lib/ansible/modules/files/synchronize.py E325
 lib/ansible/modules/files/synchronize.py E327
 lib/ansible/modules/files/unarchive.py E323
 lib/ansible/modules/identity/cyberark/cyberark_user.py E324
+lib/ansible/modules/identity/ipa/ipa_config.py E325
```
Comment (mattclay) [inline test/sanity/validate-modules/ignore.txt]: New ignores should not be added. The underlying issues should be fixed in the code instead. If the issues are in common code, such as module_utils, then those issues should be fixed first in a separate PR.

Comment (fxfitz) [reply test/sanity/validate-modules/ignore.txt]: Thanks @mattclay! I'll open up a new PR for fix all of the ipa_ issues, and then rebase this one to fix it.

Comment (fxfitz) [reply test/sanity/validate-modules/ignore.txt]: I just opened https://github.com/ansible/ansible/pull/42282, so I'll wait to get that merged and then I'll rebase this PR with the appropriate changes.

File: lib/ansible/module_utils/ipa.py [hunk 2]
```diff
@@ -97,10 +97,14 @@ def _post_json(self, method, name, item=None):
             item = {}
         url = '%s/session/json' % self.get_base_url()
         data = dict(method=method)
-        if method != 'ping':
-            data['params'] = [[name], item]
-        else:
+
+        # TODO: We should probably handle this a little better.
```
Comment (Akasurde) [inline lib/ansible/module_utils/ipa.py:101]: Could you please elaborate more about this ?

File: lib/ansible/modules/identity/ipa/ipa_config.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: ipa_config
+author: Fran Fitzpatrick (@fxfitz)
+short_description: Manage Global FreeIPA Configuration Settings
+description:
+- Modify global configuration settings of a FreeIPA Server
```
Comment (Akasurde) [inline lib/ansible/modules/identity/ipa/ipa_config.py]: Period at end.

File: lib/ansible/modules/identity/ipa/ipa_config.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+options:
+  ipadefaultloginshell:
+    description: Default shell for new users
```
Comment (Akasurde) [inline lib/ansible/modules/identity/ipa/ipa_config.py]: idem

File: lib/ansible/modules/identity/ipa/ipa_config.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+version_added: "2.7"
+'''
+
+EXAMPLES = '''
+# Ensure the default login shell is bash
+- ipa_config:
+    ipadefaultloginshell: /bin/bash
+    ipa_host: localhost
+    ipa_user: admin
+    ipa_pass: supersecret
```
Comment (Akasurde) [inline lib/ansible/modules/identity/ipa/ipa_config.py:38]: Add another example with `ipadefaultemaildomain`

Comment (fxfitz) [reply lib/ansible/modules/identity/ipa/ipa_config.py:38]: Corrected.

File: lib/ansible/modules/identity/ipa/ipa_config.py (hunk 4 above)
Comment (fxfitz) [reply lib/ansible/modules/identity/ipa/ipa_config.py]: @Akasurde Sorry, what are you referring to here?

File: lib/ansible/module_utils/ipa.py (hunk 2 above)
Comment (fxfitz) [reply lib/ansible/module_utils/ipa.py:101]: @Akasurde So when I was adding this module, I found it surprising that I needed to make the change here in `module_utils/ipa.py`. I thought that _maybe_ we could do a better job with the function implementation so that we didn't have to add "exceptions"/"switch statements" for certain API methods like how we're doing now.

I honestly didn't think too much more about it than that, and just added that comment. :-)

**Comment by Akasurde** [comment]
@fxfitz Looks good to me. Are you planning to add other configuration settings in this same PR or next iteration ?

**Comment by fxfitz** [comment]
@Akasurde Just added the aliases! Thanks! :-D

**Comment by Akasurde** [comment]
rebuild_merge

**Comment by Akasurde** [comment]
@fxfitz Thanks for the contribution.
